src/main/java/net/snowflake/client/jdbc/SnowflakeBaseResultSet.java

```diff
@@ -1721,4 +1741,43 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
     return iface.isInstance(this);
   }
+
+  private Date convertToDate(Object object, TimeZone tz) throws SFException {
+    if (sfBaseResultSet instanceof SFArrowResultSet) {
+      return sfBaseResultSet
+          .getConverters()
+          .getStructuredTypeDateTimeConverter()
+          .getDate((int) object, tz);
+    } else {
+      return (Date) sfBaseResultSet.getConverters().dateConverter(session).convert((String) object);
+    }
+  }
+
+  private Time convertToTime(Object object, TimeZone tz, int scale) throws SFException {
+    if (sfBaseResultSet instanceof SFArrowResultSet) {
+      return sfBaseResultSet
+          .getConverters()
+          .getStructuredTypeDateTimeConverter()
+          .getTime((long) object, scale);
+    } else {
+      return (Time) sfBaseResultSet.getConverters().timeConverter(session).convert((String) object);
+    }
+  }
+
+  private Timestamp convertToTimestamp(
+      Object object, int columnType, int columnSubType, TimeZone tz, int scale) throws SFException {
+    if (sfBaseResultSet instanceof SFArrowResultSet) {
```

Comment:
I couldn't be checked here. Probably it is wrong that the method for JSONs was implemented in Base..... Could you take a look?

Reply:
Yes, refactored it to generic sfBaseResultSet methods.